### PR TITLE
feat(proxy): In app full proxy

### PR DESCRIPTION
**Pull request by Champii**

### CODE REVIEW

File: src/proxy/inapp/inapp_full_proxy.go [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+		stateHash:             []byte{},
+		committedTransactions: [][]byte{},
+		logger:                logger,
+		snapshotRequestCh:     make(chan bproxy.SnapshotRequest),
+		restoreCh:             make(chan bproxy.RestoreRequest),
+		timeout:               timeout,
+	}
+}
+
+func (p *InmemFullProxy) commit(block hg.Block) []byte {
```
Comment (Champii) [inline src/proxy/inapp/inapp_full_proxy.go]: I don't know if this is relevant anymore

Comment (arrivets) [reply src/proxy/inapp/inapp_full_proxy.go]: Yes it is not necessary anymore

File: src/proxy/inapp/inapp_full_proxy.go [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+	logger                *logrus.Logger
+	snapshotRequestCh     chan bproxy.SnapshotRequest
+	restoreCh             chan bproxy.RestoreRequest
+	timeout               time.Duration
+}
+
+// NewInmemFullProxy instantiate an InmemFullProxy.
+// If no logger, a new one is created
+func NewInmemFullProxy(timeout time.Duration, logger *logrus.Logger) *InmemFullProxy {
+	if logger == nil {
```
Comment (arrivets) [inline src/proxy/inapp/inapp_full_proxy.go]: How does one provide/retrieve the CommitCh or SubmitCh?
Actually, for mobile bindings (Java, ObjectiveC), we cannot use go channels.
We need to use callbacks. Check out the mobile app proxy.

File: src/proxy/inapp/inapp_full_proxy.go [hunk 3]
```diff
@@ -0,0 +1,188 @@
+package app
```
Comment (arrivets) [inline src/proxy/inapp/inapp_full_proxy.go]: This file and the test file use different package names. 
Should probably be inapp. Doesnt compile.

File: src/proxy/inapp/inapp_full_proxy_test.go [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestInappFullProxyBabbleSide(t *testing.T) {
+	proxy := NewInmemFullProxy(1*time.Second, common.NewTestLogger(t))
+
+	state := p.State
+
```
Comment (arrivets) [inline src/proxy/inapp/inapp_full_proxy_test.go]: Doesnt compile

File: src/proxy/inapp/inapp_full_proxy.go (hunk 2 above)
Comment (Champii) [reply src/proxy/inapp/inapp_full_proxy.go]: Fixed

File: src/proxy/inapp/inapp_full_proxy_test.go (hunk 4 above)
Comment (Champii) [reply src/proxy/inapp/inapp_full_proxy_test.go]: Fixed

**Pull request closed without merging**